src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Guid.cs

```diff
@@ -108,16 +114,24 @@ private static bool TryParseGuidCore(ReadOnlySpan<byte> source, bool ends, char
                 return false;
             }
 
-            if (ends)
+            // The 'ends' parameter is a 16-bit value where the byte denoting the starting
+            // brace is at byte position 0 and the byte denoting the closing brace is at
+            // byte position 1. If no braces are expected, has value 0.
+            // Default: ends = 0
+            //  Braces: ends = "}{"
+            //  Parens: ends = ")("
+
+            if (ends != default)
             {
-                if (source[0] != begin)
+                if (source[0] != (byte)ends)
```

Comment:
The JIT could produce better codegen for this pattern. See https://github.com/dotnet/runtime/issues/33504.